### DISCUSSION TITLE
Overriding the DetailAdmin view for a content type with a list part.

### DISCUSSION THREAD

**Discussion by DrewBrasher** [OrchardCMS/OrchardCore#11575] [Q&A] [answered]
I've created an admin theme that uses TheAdmin theme as the base theme. I have a content type called BoardPacket that has a title and a ListPart. I want to add some additional functionality to the admin page where you add and remove list part items so I added a view called `Content-BoardPacket.DetailAdmin.cshtml`. The view works to display the title and some additional html but I can't figure out how to render the ListPart. I've tried different variations of the code below but here is the latest I've tried.  I'm sure it's something simple but I can't figure out what I'm doing wrong.
```
@inject OrchardCore.ContentManagement.Display.IContentItemDisplayManager ContentItemDisplayManager
@inject OrchardCore.DisplayManagement.ModelBinding.IUpdateModelAccessor UpdateModelAccessor

<h1>@Model.ContentItem.DisplayText</h1>
<h2>This is a razor test</h2>
@{
var listPart = Model.ContentItem.Content.ListPart;
var itemContent = (dynamic)await ContentItemDisplayManager.BuildDisplayAsync(listPart, UpdateModelAccessor.ModelUpdater); 
}
@await DisplayAsync(itemContent)
```

Edit:
Below is a screenshot of what I'm wanting the end result to look like. I'm wanting to add the  "Do Something" button.
![image](https://user-images.githubusercontent.com/9272802/164738201-615b4878-2361-4beb-ac01-7d9f5cd9ee3a.png)

**Comment by ns8482e** [comment]
You need to build display using display type "DetailAdmin" 
Also BuildDisplayAsync expects ContentItem not Part .

**Reply by DrewBrasher** [reply]
> Also BuildDisplayAsync expects ContentItem not Part .

Should I be using something else then to render the ListPart? OrchardCore.Lists has a view called `ListPartDetailAdmin.cshtml` which is what I'm wanting to render after the BoardPacket display text and my own html.

**Reply by ns8482e** [reply]
Create view BoardPacket-[PartName].DetailAdmin.cshtml

**Reply by DrewBrasher** [reply]
I'm not wanting to override the list part view, I want the default view used for the ListPart. I want to override the view for the content item that contains the listpart which I was able to do but now the list part is not rendered at all so I'm trying to figure out what to put in the view for the content item that will render the list part with the default view (ListPartDetailAdmin.cshtml).

**Reply by ns8482e** [reply]
Ok, then in `Content-BoardPacket.DetailAdmin.cshtml` render`ListPart` shape.

```cs
var partShape = Model.Content.PartName;
@await DisplayAsync(partShape)
```

or Render all `Content` Zone to render all parts and fields - `@await DisplayAsync(Model.Content)`

**Reply by DrewBrasher** [reply]
I tried both of those and the ListPart still does not get rendered. It looks like this:
![image](https://user-images.githubusercontent.com/9272802/164723110-f4f1c0bf-bcc3-4fc6-9ffe-46ed356629a4.png)

The partShape is null:

![image](https://user-images.githubusercontent.com/9272802/164724943-c8cdaae3-092d-49b8-8a85-174af745013a.png)

**Reply by Skrypt** [reply]
Try with Model.Items. It seems like your model is already a ListPart.
So something like 

```C#
for each (var item in Model.Items)
{
    @await DisplayAsync(item);
}
```

**Reply by DrewBrasher** [reply]
The list items are still not rendered. I added a screenshot to my original question that shows what I'm trying to do just to make sure I'm explaining it well.

**Reply by ns8482e** [reply]
are you trying to use it in dashboard?

**Reply by DrewBrasher** [reply]
Yes. I'm trying to add something to the admin dashboard

**Reply by Skrypt** [reply]
Here as a reference : 

https://github.com/OrchardCMS/OrchardCore/blob/9f963e0ebccb562aa6545497eb84100844cc24d5/src/OrchardCore.Modules/OrchardCore.Lists/Views/ListPart.cshtml#L8-L15

You need to build a shape with the ContentItem before calling DisplayAsync().

**Comment by ns8482e** [comment, marked as answer]
@DrewBrasher I guess list part is an exceptional case where it's not just `PartName` but it also consider Shape ` Type`  for naming the shape. As  List can have more than one kind of shape types. So it would be named as `[PartName]-[ShapeType]`  See https://docs.orchardcore.net/en/latest/docs/reference/modules/Templates/#razor

Also if you are using `ContentItem` in dashboard then `Content-BoardPacket.DetailAdmin.cshtml` is invalid template - you need to name it as `DashboardWidget-BoardPacket.DetailAdmin.cshtml` instead.

Inside `DashboardWidget-BoardPacket.DetailAdmin.cshtml` you can reference `ListPart` shape as following
```cs
 var  partShape  = Model.Content.Named("ListPart-ListPartDetailAdmin")`;
```

**Reply by DrewBrasher** [reply]
That worked! Thank you @ns8482e and @Skrypt for helping me out with this.
